spec/models/room_spec.rb

```diff
@@ -54,6 +54,111 @@
     end
   end
 
+  describe 'private methods' do
+    describe '#generate_code' do
+      before do
+        allow_any_instance_of(described_class).to receive(:generate_code).and_call_original
```

Comment:
Some may not like this because they believe  a private method should remain private even in tests and only functional test it through public methods that uses it.

While others will simply say a private method isn't that private after all, if not suspect to frequent changes in its API and unit testing it is possible without introducing a lot of complexity it may make sense and doesn't do any harm.

Don't know what school you like guys 😆 
Tell me what you think,

@farhatahmad @scouillard @hadicheaito1